### PR TITLE
[Kotlin Migration] GestureMapper

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/GestureMapper.kt [hunk 1]
```diff
@@ -13,185 +13,143 @@
...
-    private static TriState clamp(double value) {
-        double val = Math.floor(value);
-        if (val >= 2) {
-            return TriState.HIGH;
-        }
-        if (val < 1) {
-            return TriState.LOW;
+        // clamps the value from LOW-MID-HIGH
+        private fun clamp(value: Double): TriState {
+            val `val` = floor(value)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/GestureMapper.kt:146]: cleanup: val

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
one nit

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.kt [hunk 2]
```diff
@@ -82,7 +82,7 @@ class GestureProcessor(private val processor: ViewerCommand.CommandProcessor?) {
 
     fun onFling(dx: Float, dy: Float, velocityX: Float, velocityY: Float, isSelecting: Boolean, isXScrolling: Boolean, isYScrolling: Boolean): Boolean? {
         val gesture = gestureMapper.gesture(dx, dy, velocityX, velocityY, isSelecting, isXScrolling, isYScrolling)
-        return execute(gesture)
+        return execute(gesture!!)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.kt:85]: `gesture` can be null, the method likely wants to accept a nullable

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.kt (hunk 2 above)
Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.kt:85]: changes are done by reference from previous [java file](https://github.com/ankidroid/Anki-Android/blob/e5405ed2ec19090858997ac68b2ce6f09a9c2d58/AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.java#L128)

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.kt]: https://github.com/ankidroid/Anki-Android/blob/e5405ed2ec19090858997ac68b2ce6f09a9c2d58/AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.java#L105-L108

To my understanding, Gesture is nullable here

`mapGestureToCommand` returns `COMMAND_NOTHING`

Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/GestureProcessor.kt]: yes did the changes

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**